subsys/net/lib/openthread/platform/shell.c

```diff
@@ -27,6 +27,10 @@ int otConsoleOutputCallback(void *aContext, const char *aFormat,
 {
 	ARG_UNUSED(aContext);
 
+	if (shell_p == NULL) {
+		return 0;
```

Comment:
Return some error here?

Reply:
Actually, I think this is fine as the callback documentation says it should return `Number of bytes written by the callback.`. OT uses positive values for error codes, so I'm not sure what error should be returned in such case.

I'm a little more concerned about the return value when the function succeeds, because it's not possible to obtain the actual number of bytes sent by `shell_vfprintf` as it doesn't return anything. From what I've seen though OT does not verify the result in any way, so it wasn't an issue so far.